src/utils/memoFn.js

```diff
@@ -0,0 +1,19 @@
+// let cachedResult = 0;
+
+export const memoFn = fn => {
+  const cache = {};
+
+  return (...args) => {
+    const key = args.join('-');
+
+    if (cache[key]) {
```

Comment:
This means the cache can never store falsy values. Should be `if (key in cache)` or `if (cache.hasOwnProperty(key)`. Or maybe the cache should use a `Set`?